core/consensus/aura/src/lib.rs

```diff
@@ -496,7 +499,21 @@ fn check_header<B: Block, P: Pair>(
 		let public = expected_author;
 
 		if P::verify(&sig, &to_sign[..], public) {
-			Ok(CheckedHeader::Checked(header, digest_item))
+			match check_equivocation::<_, _, <P as Pair>::Public>(client, slot_num, header.clone(), public.clone()) {
+				Ok(Some(equivocation_proof)) => {
+					// TODO: dispatch report here.
+					Err(format!("Slot author is equivocating with headers {:?} and {:?}",
```

Comment:
We should log this with severity at least `warn`, as it should not happen normally.

Reply:
I added it as `info`.